### ISSUE TITLE
Previewer for 360 panoramic images of experiments and CERN

### ISSUE THREAD

**Issue by tpmccauley** [cernopendata/opendata.cern.ch#1441]
It might be nice to have some content on the portal like found here:

http://cern.ch/cms-aframe

where on the desktop one can view the 360 (4pi really) panoramic images available on CDS as a "magic window". If viewed on a mobile device this works as well but even better can be viewed in 3D with a Google Cardboard (e.g.)

**Comment by katilp** [comment]
This would be great to have, together with event display in 3D. I can say from experience that this would be really appreciated by teachers and students! The link should  be in a prominent link on mobile devices, @sefeg @tiborsimko referring to discussion on gitter (maybe cross-link this to another issue if such already exists

**Comment by katilp** [comment]
We could have a "Go3D" link bringing to a page(/story/guide/template as you wish) with a collection of Google Cardboard stuff

**Comment by sefeg** [comment]
@katilp I agree, it would definitely be appreciated! One could also imagine to add a tag "3D/VR" to the corresponding records. That would nicely fit into the targeted UX, as users could then explore and filter such content.

**Comment by tpmccauley** [comment]
With A-Frame the code for a "magic window" is simple. An example is here:
where there is actually a bit added to do the drag and drop. However, it
can be as simple as specifying a file name in the element.

https://github.com/tpmccauley/panorama-viewer

https://tpmccauley.github.io/panorama-viewer/

**Comment by tpmccauley** [comment]
@katilp @sefeg Where is the best place to make this sort of content available? For example, I could imagine something like this:

https://tpmccauley.github.io/lhc-panoramas/

(i.e. the frame with panoramic image)

on 

http://opendata-dev.web.cern.ch/about/cms

between "About CMS" above "About CMS data". Similarly for the other experiments as well?

**Comment by tpmccauley** [comment]
I made a prototype for CMS at about/CMS; is this a good location?
<img width="1006" alt="screen shot 2017-11-06 at 4 17 37 pm" src="https://user-images.githubusercontent.com/496503/32451367-8678a35c-c30e-11e7-922a-469fed4a9b10.png">

**Comment by tpmccauley** [comment]
On the desktop when one clicks on the "goggles" in the lower right the view renders nicely in full-screen and one can rotate around the panorama. "esc" gets one back to regular view. I have not tried on mobile :)

**Comment by katilp** [comment]
Nice! This could be a good location, we will need to check when #1518 becomes available,  as in absence of fixed entry page we will need to guide users to CMS resources they need/are interested in.

**Comment by tpmccauley** [comment]
One reason I was thinking of /about/* was that each experiment could have its own image. And an image of the LHC tunnel could go at /about

**Comment by tpmccauley** [comment]
I have added aframe on a branch so it ready to use. Where shall we add CMS panorama to start? It looks like the about CMS page is md so am wondering how to add js and html if we want to include aframe on this page.

**Comment by RaoOfPhysics** [comment]
@tpmccauley: I think .md pages can generally include HTML code, but not sure what translator we're using here. Can you try on a local instance?

**Comment by tpmccauley** [comment]
It seems that html is ok since paragraph text gets wrapped in p tags.
However don't know how to include assets from bundles like aframe js.
@tiborsimko?

On 30 Nov 2017 16:56, "Achintya Rao" <notifications@github.com> wrote:

@tpmccauley <https://github.com/tpmccauley>: I think .md pages can
generally include HTML code, but not sure what translator we're using here.
Can you try on a local instance?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/cernopendata/opendata.cern.ch/issues/1441#issuecomment-348251065>,
or mute the thread

**Comment by tpmccauley** [comment]
Is this worth revisiting for new release?

**Comment by tiborsimko** [comment]
For the the assets, you can try amending `cernopendata/modules/theme/bundles.py` to include whatever is necessary... You could also grep how `demobbed-viewer` is used for OPERA event display
